### ISSUE TITLE
cifar10 default graph runtime error

### ISSUE THREAD

**Issue by mrrosen** [pytorch/glow#4158]
When trying to run the cifar10 example, I ran into the same problem as this poster when they were running the mnist example:

https://discuss.pytorch.org/t/glow-example-run-error/50461

The same fix worked, by changing all the connections from MaxPool layers to use the `getResult` method. I dont know enough about glow to understand what the difference is as all the other layers are connected "directly" by referencing the object output from `createXXX` while the MaxPool ones need this `getResult`.

Changes that fixed the problem:

    --- a/examples/cifar10.cpp
    +++ b/examples/cifar10.cpp
    @@ -65,15 +65,15 @@ static Placeholder *createDefaultModel(PlaceholderBindings &bindings,
       auto *RL0 = F->createRELU("relu", CV0);
       auto *MP0 = F->createMaxPool("pool", RL0, 2, 2, 0);
    
    -  auto *CV1 = F->createConv(bindings, "conv", MP0, 20, 5, 1, 2, 1);
    +  auto *CV1 = F->createConv(bindings, "conv", MP0->getResult(), 20, 5, 1, 2, 1);
       auto *RL1 = F->createRELU("relu", CV1);
       auto *MP1 = F->createMaxPool("pool", RL1, 2, 2, 0);
    
    -  auto *CV2 = F->createConv(bindings, "conv", MP1, 20, 5, 1, 2, 1);
    +  auto *CV2 = F->createConv(bindings, "conv", MP1->getResult(), 20, 5, 1, 2, 1);
       auto *RL2 = F->createRELU("relu", CV2);
       auto *MP2 = F->createMaxPool("pool", RL2, 2, 2, 0);
    
    -  auto *FCL1 = F->createFullyConnected(bindings, "fc", MP2, numLabels);
    +  auto *FCL1 = F->createFullyConnected(bindings, "fc", MP2->getResult(), numLabels);
       auto *SM = F->createSoftMax("softmax", FCL1, expected);
       auto *save = F->createSave("ret", SM);
       return save->getPlaceholder();

**Comment by jfix71** [comment]
Thanks for pointing this out @mrrosen -- would you like to put up a PR to fix this?

FYI, the reason this happens is that MaxPool has multiple results. We have the concept of a `Node` vs a `NodeValue` -- a NodeValue is essentially a wrapper around a Node and an unsigned integer representing which specific result number you're referring to from it. If you do not call `get*()` to get a specific `NodeValue` result from a Node with multiple results then it will default to using the 0th result in Release, and it will cause an assertion failure in Debug. If a Node has a single result then you do not need to `get*()` to get a NodeValue as it will default to the only result the Node has.

**Comment by mrrosen** [comment]
@jfix71 

Thanks for the explanation, Ive created a PR for the change I made above
